activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -86,6 +86,20 @@ def supports_lazy_transactions?
         true
       end
 
+      def column_definitions(table_name)
+        if !mariadb? && get_database_version > "8.0.2"
+          execute_and_free("SHOW FULL FIELDS FROM #{quote_table_name(table_name)}", "SCHEMA") do |result|
+            each_hash(result).each do |col_info|
+              if col_info[:Type].start_with?("varbinary") && col_info.fetch(:Default)&.start_with?("0x")
+                col_info[:Default] = [ col_info[:Default][2...]].pack("H*")
+              end
+            end
+            each_hash(result)
```

Comment:
Calling `each_hash` multiple times feels a bit odd (Depends on the internals that the mutations to hashes done above).

Do you think we could call `super` here and then mutate the result? Like `super.tap do ...`